skopt/wrappers/search_cv.py

```diff
@@ -0,0 +1,490 @@
+             estimator=estimator, scoring=scoring, fit_params=fit_params,
+             n_jobs=n_jobs, iid=iid, refit=refit, cv=cv, verbose=verbose,
+             pre_dispatch=pre_dispatch, error_score=error_score,
+             return_train_score=return_train_score)
+
+    def _fit_best_model(self, X, y):
+        """Fits the estimator copy with best parameters found to the
+        provided data.
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+            Input data, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape = [n_samples] or [n_samples, n_output],
+            Target relative to X for classification or regression.
+
+        Returns
+        -------
+        self
+        """
+        self.best_estimator_ = clone(self.estimator)
+        self.best_estimator_.set_params(**self.best_params_)
+        self.best_estimator_.fit(X, y)
+        return self
+
+    def _make_optimizer(self, params_space):
+        """Method to instantiate skopt Optimizer class.
+
+        Parameters
+        ----------
+        params_space : dict
+            Represents parameter search space. The keys are parameter
+            names (strings) and values are skopt.space.Dimension instances,
+            one of Real, Integer or Categorical.
+
+        y : array-like, shape = [n_samples] or [n_samples, n_output],
+            Target relative to X for classification or regression.
+
+        Returns
+        -------
+        optimizer: Instance of the `Optimizer` class used for for search
+            in some parameter space.
+
+        """
+        # convert search space from dict to list
+        dimensions = [params_space[k] for k in sorted(params_space.keys())]
+
+        if self.surrogate == "auto":
+            surrogate = GaussianProcessRegressor()
+        else:
+            surrogate = self.surrogate
+
+        optimizer = Optimizer(dimensions, surrogate, acq_optimizer='sampling')
+        return optimizer
+
+    def _skopt_to_dict(self, params_space, params):
+        """Converts list of parameter values into the dictionary with
+        keys as parameter names and corresponding values of parameter.
+
+        Parameters
+        ----------
+        params_space : dict
+            Represents parameter search space. The keys are parameter
+            names (strings) and values are skopt.space.Dimension instances,
+            one of Real, Integer or Categorical.
+
+        params : list
+            Parameter values as list. The order of parameters in the list
+            is given by sorted(params_space.keys()).
+
+        Returns
+        -------
+        params_dict: dictionary with parameter values.
+        """
+        params_dict = {
+            k: v for k, v in zip(sorted(params_space.keys()), params)
+        }
+        return params_dict
+
+    def step(self, X, y, param_space, groups=None, n_jobs=1):
```

Comment:
I'd have to read the code for `RandomSearchCV` but doesn't it have a step function we can re-use/hook into so we don't have to duplicate all the code for recording the results?